dev/com.ibm.ws.security.openidconnect.clients.common/test/com/ibm/ws/security/openidconnect/clients/common/UserInfoHelperTest.java

```diff
@@ -0,0 +1,169 @@
+
+    class UserInfoHelperMock extends UserInfoHelper {
+        String mockURLResponse = USERINFO;
+        String mockUpdateResult = null;
+        boolean updateAuthResultCalled = false;
+
+        public UserInfoHelperMock(ConvergedClientConfig config) {
+            super(config);
+        }
+
+        @Override
+        protected String getUserInfoFromURL(ConvergedClientConfig config, SSLSocketFactory sslsf, String accessToken) {
+            return mockURLResponse;
+        }
+
+        @Override
+        protected void updateAuthenticationResultPropertiesWithUserInfo(ProviderAuthenticationResult oidcResult, String userInfoStr) {
+            mockUpdateResult = userInfoStr;
+            updateAuthResultCalled = true;
+        }
+
+    }
+
+    @Before
+    public void before() {
+
+    }
+
+    @After
+    public void after() {
+        mock.assertIsSatisfied();
+    }
+
+    void setExpectations() {
+        mock.checking(new Expectations() {
+            {
+                allowing(convClientConfig).getUserInfoEndpointUrl();
+                will(returnValue("somebogusurl"));
+                allowing(convClientConfig).isUserInfoEnabled();
+                will(returnValue(true));
+            }
+        });
+    }
+
+    @Test
+    public void testRetrieveValidInfo() {
+        setExpectations();
+        UserInfoHelperMock uihm = new UserInfoHelperMock(convClientConfig);
+        boolean result = uihm.getUserInfo(null, null, "accesstoken", "testuser");
+        assertTrue("method return should be true", result);
+        assertTrue("userinfo result" + uihm.mockUpdateResult + " did not match expected value: " + uihm.mockURLResponse,
+                uihm.mockUpdateResult.equals(uihm.mockURLResponse));
+    }
+
+    /**
+     * retrieve userinfo where subject from id token doesn't match what's in userinfo.
+     * userinfo should be ignored, updateAuthResult should not be called
+     */
+    @Test
+    public void testRetrieveInvalidInfo() {
+        setExpectations();
+        UserInfoHelperMock uihm = new UserInfoHelperMock(convClientConfig);
+        // put the wrong sub in the userinfo, should not get the userinfo back.
+        uihm.mockURLResponse = uihm.mockURLResponse.replace("testuser", "bogususer");
+        boolean result = uihm.getUserInfo(null, null, "accesstoken", "testuser");
+        assertFalse("method return should be false", result);
+        assertFalse("updateAuthResult should not have been called",
+                uihm.updateAuthResultCalled);
+    }
+
+    /**
+     * retrieve userinfo where no subject from id token is present.
+     * userinfo should be ignored, since id token subject and userinfo subject don't match.
+     */
+    @Test
+    public void testRetrieveNoSujbectFromIdToken() {
+        setExpectations();
+        UserInfoHelperMock uihm = new UserInfoHelperMock(convClientConfig);
+        boolean result = uihm.getUserInfo(null, null, "accesstoken", null);
```

Comment:
Would suggest tests be added for:
- Config does not have a user info endpoint configured or user info is disabled by the config
- Access token is null